src/async_fuse/memfs/dist/server.rs

```diff
@@ -76,62 +57,64 @@ impl CacheServer {
             .parse()
             .unwrap_or_else(|e| panic!("Failed to parse ip {}, error is {}", ip, e));
 
-        let th = thread::spawn(move || {
-            let listener =
-                TcpListener::bind(format!("{}:{}", ip_copy, port_copy)).unwrap_or_else(|e| {
-                    panic!(
-                        "Fail to bind tcp listener to {}:{}, error is {}",
-                        ip_copy, port_copy, e
-                    )
-                });
-            loop {
-                match listener.accept() {
-                    Ok((stream, addr)) => {
-                        // Receive connection from local means to turnoff server.
-                        if addr.ip() == ip_addr {
-                            let mut buf = Vec::new();
-                            let mut local_stream = stream;
-                            if let Err(e) = tcp::read_message(&mut local_stream, &mut buf) {
-                                panic!(
-                                    "fail to read distributed cache request from tcp stream, {}",
-                                    e
-                                );
-                            }
+        let listener_join_handler = tokio::spawn(listen(ip_copy, port_copy, ip_addr, cache, meta));
+        Self {
+            ip,
+            port,
+            listener_join_handler,
+        }
+    }
+}
 
-                            let request = request::deserialize_cache(buf.as_slice());
-                            if let DistRequest::TurnOff = request {
-                                turnoff(&mut local_stream).unwrap_or_else(|e| {
-                                    panic!("failed to send turnoff reply, error is {}", e)
-                                });
-                                return true;
-                            }
-                            panic!(
-                                "should only receive turnoff request from local, request is {:?}",
-                                request
-                            );
-                        } else {
-                            let cache_clone = Arc::<GlobalCache>::clone(&cache);
-                            let meta_clone = Arc::<S3MetaData<S>>::clone(&meta);
+/// async listen routine
+async fn listen<S: S3BackEnd + Send + Sync + 'static>(
+    ip: String,
+    port: String,
+    ip_addr: IpAddr,
+    cache: Arc<GlobalCache>,
+    meta: Arc<S3MetaData<S>>,
+) {
+    let listener = tokio::net::TcpListener::bind(format!("{}:{}", ip, port))
+        .await
+        .unwrap_or_else(|e| {
+            panic!(
+                "Fail to bind tcp listener to {}:{}, error is {}",
+                ip, port, e
+            );
+        });
+    loop {
+        match listener.accept().await {
+            Ok((stream, addr)) => {
+                // Receive connection from local means to turnoff server.
+                if addr.ip() == ip_addr {
```

Comment:
This if branch is not needed any more.

Reply:
this if branch might prevent the system from recving anything from localhost

Reply:
original purpose if this branch is to turn off this task. Ok to remove because we will never receive message from local anymore. Even through we receive, it is ok to handle.

Reply:
done